lib/mix/lib/mix/dep/lock.ex

```diff
@@ -32,48 +32,78 @@ defmodule Mix.Dep.Lock do
   """
   @spec read() :: map
   def read() do
-    case File.read(lockfile()) do
-      {:ok, info} ->
-        assert_no_merge_conflicts_in_lockfile(lockfile(), info)
+    lockfile = lockfile()
 
-        case Code.eval_string(info, [], file: lockfile()) do
-          {lock, _binding} when is_map(lock) -> lock
-          {_, _binding} -> %{}
-        end
+    case File.read(lockfile) do
+      {:ok, content} ->
+        read_lock(content, lockfile)
 
       {:error, _} ->
         %{}
     end
   end
 
+  @conflict_delimiters ["<<<<<<<", "=======", ">>>>>>>"]
+
+  defp read_lock(content, lockfile) do
+    read_lock(content, lockfile, String.contains?(content, @conflict_delimiters))
```

Comment:
I don't see the point in doing `read_lock/3` with a boolean as the third argument, I find that it obfuscates the conditional a lot. Why not go with

```elixir
defp read_lock(content, lockfile) do
  if String.contains?(...) do
    ...
  else
    ...
  end
end
```

?

Reply:
I see, good point... I'll change it back to an if